decimal.go

```diff
@@ -1307,6 +1307,31 @@ func (d NullDecimal) MarshalJSON() ([]byte, error) {
 	return d.Decimal.MarshalJSON()
 }
 
+// UnmarshalText implements the encoding.TextUnmarshaler interface for XML
+// deserialization
+func (d *NullDecimal) UnmarshalText(text []byte) error {
+	str := string(text)
+
+	if str == "" {
+		d.Valid = false
+		return nil
+	}
+	if err := d.Decimal.UnmarshalText(text); err != nil {
+		return err
```

Comment:
Shouldn't we explicitly set here `d.Valid = false`?

Reply:
We are returning error. d is invalid and shouldn't be used by caller. We can set it to false, I dont have strong opinion about it.